src/Microsoft.CodeQuality.Analyzers/Microsoft.CodeQuality.Analyzers.md

```diff
@@ -0,0 +1,547 @@
+
+Category: Design
+
+Severity: Warning
+
+Help: [https://msdn.microsoft.com/en-us/library/ms182166.aspx](https://msdn.microsoft.com/en-us/library/ms182166.aspx)
+
+### CA1043: Use Integral Or String Argument For Indexers ###
+
+Indexers, that is, indexed properties, should use integer or string types for the index. These types are typically used for indexing data structures and increase the usability of the library. Use of the Object type should be restricted to those cases where the specific integer or string type cannot be specified at design time. If the design requires other types for the index, reconsider whether the type represents a logical data store. If it does not represent a logical data store, use a method.
+
+Category: Performance
+
+Severity: Warning
+
+Help: [https://msdn.microsoft.com/en-us/library/ms182180.aspx](https://msdn.microsoft.com/en-us/library/ms182180.aspx)
+
+### CA1050: Declare types in namespaces ###
+
+Types are declared in namespaces to prevent name collisions and as a way to organize related types in an object hierarchy.
+
+Category: Design
+
+Severity: Warning
+
+Help: [https://msdn.microsoft.com/en-us/library/ms182134.aspx](https://msdn.microsoft.com/en-us/library/ms182134.aspx)
+
+### CA1052: Static holder types should be Static or NotInheritable ###
+
+Category: Design
+
+Severity: Warning
+
+Help: [http://msdn.microsoft.com/library/ms182168.aspx](http://msdn.microsoft.com/library/ms182168.aspx)
+
+### CA1060: Move pinvokes to native methods class ###
+
+Platform Invocation methods, such as those that are marked by using the System.Runtime.InteropServices.DllImportAttribute attribute, or methods that are defined by using the Declare keyword in Visual Basic, access unmanaged code. These methods should be of the NativeMethods, SafeNativeMethods, or UnsafeNativeMethods class.
+
+Category: Design
+
+Severity: Warning
+
+Help: [http://msdn.microsoft.com/library/ms182161.aspx](http://msdn.microsoft.com/library/ms182161.aspx)
+
+### CA1064: Exceptions should be public ###
+
+An internal exception is visible only inside its own internal scope. After the exception falls outside the internal scope, only the base exception can be used to catch the exception. If the internal exception is inherited from T:System.Exception, T:System.SystemException, or T:System.ApplicationException, the external code will not have sufficient information to know what to do with the exception.
+
+Category: Design
+
+Severity: Warning
+
+Help: [https://msdn.microsoft.com/en-us/library/bb264484.aspx](https://msdn.microsoft.com/en-us/library/bb264484.aspx)
+
+### CA1065: Do not raise exceptions in unexpected locations ###
+
+A method that is not expected to throw exceptions throws an exception.
+
+Category: Design
+
+Severity: Warning
+
+Help: [https://msdn.microsoft.com/en-us/library/bb386039.aspx](https://msdn.microsoft.com/en-us/library/bb386039.aspx)
+
+### CA1066: Type {0} should implement IEquatable<T> because it overrides Equals ###
+
```

Comment:
No description or link? Do we not have a page for this one or the next two?

Reply:
Seems like next set of rules are not implemented by legacy FxCop, hence no msdn link. I will update these once I tackle https://github.com/dotnet/roslyn-analyzers/issues/1269

Reply:
I will add the description for these rules in this PR